dask/delayed.py

```diff
@@ -86,7 +86,7 @@ def to_task_dask(expr):
             return (type(expr), args), dsk
         else:
             return args, dsk
-    if isinstance(expr, dict):
+    if type(expr) is dict:
```

Comment:
I think it's easier to say that `dask.delayed` only traverses builtin types, and ignore any subclasses/duck-typed classes. Since `type(MyClass2()) is not dict`, then delayed shouldn't traverse inside it.